CuratedContent/WhatEveryComputerScientistShouldKnowAboutFloatingPointArithmetic.md

```diff
@@ -0,0 +1,79 @@
+# What Every Computer Scientist Should Know About Floating-Point Arithmetic
+
+<!--deck text start-->
+Floating point computations are the foundation for computational science and engineering software.
+As such, understanding the foundations of floating point data-types and operations are critical in the development of robust portable numerical software.
+<!--deck text end-->
+
+#### Contributed by [Roscoe A. Bartlett](https://github.com/bartlettroscoe)
+#### Publication date: ???
+
+Resource information | Details 
+:--- | :--- 
+Paper title | [What Every Computer Scientist Should Know About Floating-Point Arithmetic](https://doi.org/10.1145/103162.103163)
+Authors | David Goldberg
+Publication | Year 1991, Journal of Information and Software Technology, DOI: [https://dl.acm.org/doi/10.1145/103162.103163](https://dl.acm.org/doi/10.1145/103162.103163)
+
+Resource information | Details 
+:--- | :--- 
+Web Page | [What Every Computer Scientist Should Know About Floating-Point Arithmetic](https://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html)
+Authors | David Goldberg
+
+NOTE: The original [journal article](https://doi.org/10.1145/103162.103163) has been republished as a [hyperlinked HTML page](https://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html).
+
+Doing sufficiently accurate computations of infinite precision real numbers using finite precision computer approximations requires a solid understanding of floating-point formats, computational issues, and standards.
+When implementing numerical algorithms involving real-world data solving ill-conditioned mathematical problems, a single sloppy floating-point expression can break an implemented algorithm and make it impossible to compute a solution to the needed precision (or any solution at all).
+
+The paper "What Every Computer Scientist Should Know About Floating-Point Arithmetic" provides a critical foundation for understanding modern floating-point data-types and computations approximating algorithms involving real numbers.
+The issues involved must be well understood by computer hardware developers, system software developers, and numerical software developers alike in order for it to be possible to implement robust numerical algorithms in a portable way.
+
+This paper covers the basics of floating-point including:
+
+* floating-point formats,
+* rounding error (e.g. guard digits) and measures of error (e.g. machine epsilon &epsilon;, absolute error),
+* loss of basic associativity and loss of other basic properties of real arithmetic, and
+* catastrophic cancellation.
+
+In addition, the author covers IEEE floating-point standards including the IEEE 754 binary standard which addresses:
+
+* bit-wise floating-point number data-structures (i.e. significant digits, exponents, sign),
+* rounding and accuracy requirements for basic floating-point operations,
+* overflow, underflow and special quantities (i.e. (signed) zero, demormalized numbers, NaNs, (signed) infinity &infin;),
```

Comment:
is it denormalize

Reply:
Thanks!  Addressed in commit bcd4e2f264bb664f80eeaa4be26fbdde6918b46b.